docs/ovs-offload.md

```diff
@@ -0,0 +1,207 @@
+# OVS Hardware Offload
+
+The OVS software based solution is CPU intensive, affecting system performance
+and preventing fully utilizing available bandwidth. OVS 2.8 and above support
+new feature called OVS Hardware Offload which improves performance significantly.
+This feature allows to offload the OVS data-plane to the NIC while maintaining
+OVS control-plane unmodified. It is using SR-IOV technology with VF representor
+host net-device. The VF representor plays the same role as TAP devices
+in Para-Virtual (PV) setup. A packet sent through the VF representor on the host
+arrives to the VF, and a packet sent through the VF is received by its representor.
+
+## Supported Ethernet controllers
+
+The following manufacturers are known to work:
+
+- Mellanox ConnectX-5 and above
+
+## Prerequisites
+
+- Linux Kernel 5.7
+- Open vSwitch 2.13
+- iproute >= 4.12
+- sriov-device-plugin
+- multus-cni
+
+## Worker Node SR-IOV Configuration
+
+In order to enable Open vSwitch hardware offloading, the following steps
+are required. Please make sure you have root privileges to run the commands
+below.
+
+Check the Number of VF Supported on the NIC
+
+```
+cat /sys/class/net/enp3s0f0/device/sriov_totalvfs
+8
+```
+
+Create the VFs
+
+```
+echo '4' > /sys/class/net/enp3s0f0/device/sriov_numvfs
+```
+
+Verfiy the VFs are created
+
+```
+ip link show enp3s0f0
+8: enp3s0f0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP mode DEFAULT qlen 1000
+   link/ether a0:36:9f:8f:3f:b8 brd ff:ff:ff:ff:ff:ff
+   vf 0 MAC 00:00:00:00:00:00, spoof checking on, link-state auto
+   vf 1 MAC 00:00:00:00:00:00, spoof checking on, link-state auto
+   vf 2 MAC 00:00:00:00:00:00, spoof checking on, link-state auto
+   vf 3 MAC 00:00:00:00:00:00, spoof checking on, link-state auto
+```
+
+Setup the PF to be up
+
+```
+ip link set enp3s0f0 up
+```
+
+Unbind the VFs from the driver
+
+```
+echo 0000:03:00.2 > /sys/bus/pci/drivers/mlx5_core/unbind
+echo 0000:03:00.3 > /sys/bus/pci/drivers/mlx5_core/unbind
+echo 0000:03:00.4 > /sys/bus/pci/drivers/mlx5_core/unbind
+echo 0000:03:00.5 > /sys/bus/pci/drivers/mlx5_core/unbind
+```
+
+Configure SR-IOV VFs to switchdev mode
+
+```
+devlink dev eswitch set pci/0000:03:00.0 mode switchdev
+ethtool -K enp3s0f0 hw-tc-offload on
+```
+
+Bind the VFs to the driver
+
+```
+echo 0000:03:00.2 > /sys/bus/pci/drivers/mlx5_core/bind
+echo 0000:03:00.3 > /sys/bus/pci/drivers/mlx5_core/bind
+echo 0000:03:00.4 > /sys/bus/pci/drivers/mlx5_core/bind
+echo 0000:03:00.5 > /sys/bus/pci/drivers/mlx5_core/bind
+```
+
+Set hw-offload=true restart Open vSwitch
+
+```
+systemctl enable openvswitch.service
+ovs-vsctl set Open_vSwitch . other_config:hw-offload=true
+systemctl restart openvswitch.service
+```
+
+## Worker Node SR-IOV network device plugin configuration
+
+This plugin creates device plugin endpoints based on the configurations given in file `/etc/pcidp/config.json`.
```

Comment:
maybe a stupid question, but where does this file (`/etc/pcidp/config.json`) come from?